packages/clerk-js/src/ui/elements/PhoneInput/PhoneInput.tsx

```diff
@@ -37,9 +43,36 @@ export const PhoneInput = (props: PhoneInputProps) => {
   const selectedCountryOption = React.useMemo(() => {
     return countryOptions.find(o => o.country.iso === selectedIso) || countryOptions[0];
   }, [selectedIso]);
+  const dynamicPadding = selectedCountryOption.country.code.length * 5; // this is to calculate the padding of the input field depending the length of country code
 
   React.useEffect(callOnChangeProp, [cleanPhoneNumber]);
 
+  const handlePaste = (e: ClipboardEvent<HTMLInputElement>) => {
+    const inputValue = e.clipboardData.getData('text');
+    const pattern = /\+/g;
+
+    if (pattern.test(inputValue)) {
+      setHasBeenPasted(true);
+
+      const { result, selectedCountryCode } = getLongestValidCountryCode(inputValue, countryOptions);
```

Comment:
```suggestion
      const { result, selectedCountryCode } = getLongestValidCountryCode(inputValue);
```

We shouldn't use `countryOptions` here, because this is an array with options for the UI select element. This method purely deals with country data, so it should not be aware of ui-related concerns as this options array.

We should remove this param altogether, and instead simply use the country map within `phoneUtils`, as we do for the other utilities.